Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -673,6 +673,40 @@ theorem coeff_prod (f : ι → PowerSeries R) (d : ℕ) (s : Finset ι) :
     Equiv.coe_toEmbedding, Finsupp.mapRange.equiv_apply, AddEquiv.coe_toEquiv_symm,
     Finsupp.mapRange_apply, AddEquiv.finsuppUnique_symm]
 
+/-- The `n`-th coefficient of the `k`-th power of a power series.
+$$
+[x^n] P(x)^k \ = \ \sum_{n_1 + n_2 + \cdots + n_k = n} a_{n_1} a_{n_2} \cdots a_{n_k}
+$$
+-/
+lemma coeff_pow (k n : ℕ) (φ : R⟦X⟧) :
+    coeff R n (φ ^ k) = ∑ l ∈ finsuppAntidiag (range k) n, ∏ i ∈ range k, coeff R (l i) φ := by
+  have h₁ (i : ℕ) : Function.const ℕ φ i = φ := rfl
+  have h₂ (i : ℕ) : ∏ j ∈ range i, Function.const ℕ φ j = φ ^ i := by
+    apply prod_range_induction (fun _ => φ) (fun i => φ ^ i) rfl (congrFun rfl) i
+  rw [← h₂, ← h₁ k]
+  apply coeff_prod (f := Function.const ℕ φ) (d := n) (s := range k)
+
+/-- First coefficient of the product of two power series. -/
+lemma coeff_one_mul (φ ψ : R⟦X⟧) : coeff R 1 (φ * ψ) =
+    coeff R 1 φ * constantCoeff R ψ + coeff R 1 ψ * constantCoeff R φ := by
+  have : Finset.antidiagonal 1 = {(0, 1), (1, 0)} := by exact rfl
+  rw [coeff_mul, this, Finset.sum_insert, Finset.sum_singleton, coeff_zero_eq_constantCoeff,
+    mul_comm, add_comm]
+  norm_num
+
+/-- First coefficient of the `n`-th power of a power series with constant coefficient 1. -/
```

Comment:
Maybe the same one with an arbitrary constant coefficient could be useful.

Reply:
Appears to be `n * a1^(n-1)`. I will work on this next.